File: test-transformations.js
const axios = require('axios');

const BASE_URL = 'https://textforge.co';

const tests = [
  // String formatting
  { op: 'slugify', input: 'Hello World!', expected: 'hello-world' },
  { op: 'camelcase', input: 'hello world', expected: 'helloWorld' },
  { op: 'snakecase', input: 'Hello World', expected: 'hello_world' },
  { op: 'kebabcase', input: 'Hello World', expected: 'hello-world' },
  { op: 'pascalcase', input: 'hello world', expected: 'HelloWorld' },
  { op: 'constantcase', input: 'hello world', expected: 'HELLO_WORLD' },
  { op: 'capitalize', input: 'hello world', expected: 'Hello world' },
  { op: 'titlecase', input: 'hello world', expected: 'Hello World' },
  { op: 'sentencecase', input: 'HELLO WORLD', expected: 'Hello world' },
  
  // String manipulation
  { op: 'reverse', input: 'hello', expected: 'olleh' },
  { op: 'uppercase', input: 'hello', expected: 'HELLO' },
  { op: 'lowercase', input: 'HELLO', expected: 'hello' },
  { op: 'trim', input: '  hello  ', expected: 'hello' },
  { op: 'truncate', input: 'hello world', params: { length: 5 }, expected: 'hello...' },
  
  // Encoding/decoding
  { op: 'base64encode', input: 'hello', expected: 'aGVsbG8=' },
  { op: 'base64decode', input: 'aGVsbG8=', expected: 'hello' },
  { op: 'urlencode', input: 'hello world', expected: 'hello%20world' },
  { op: 'urldecode', input: 'hello%20world', expected: 'hello world' },
  { op: 'htmlencode', input: '<div>', expected: '&lt;div&gt;' },
  { op: 'htmldecode', input: '&lt;div&gt;', expected: '<div>' },
  
  // Analysis
  { op: 'wordcount', input: 'hello world foo', expected: 3 },
  { op: 'charcount', input: 'hello', expected: 5 },
  { op: 'linecount', input: 'line1\nline2\nline3', expected: 3 },
  
  // Extraction
  { op: 'extractemails', input: 'Contact us at test@example.com or admin@example.org', expected: ['test@example.com', 'admin@example.org'] },
  { op: 'extracturls', input: 'Visit https://example.com or http://test.org', expected: ['https://example.com', 'http://test.org'] },
  { op: 'extractnumbers', input: 'I have 5 apples and 10 oranges', expected: [5, 10] },
  
  // Cleaning
  { op: 'removehtml', input: '<p>Hello</p>', expected: 'Hello' },
  { op: 'removespecial', input: 'hello!@#$world', expected: 'helloworld' },
];

async function runTests() {
  console.log('Testing all 28 transformations...\n');
  
  let passed = 0;
  let failed = 0;
  
  for (const test of tests) {
    try {
      const params = { input: test.input, operation: test.op };
      if (test.params) Object.assign(params, test.params);
      
      const response = await axios.get(`${BASE_URL}/v1/run`, { params });
      const result = response.data.result;
      
      const success = JSON.stringify(result) === JSON.stringify(test.expected);
      
      if (success) {
        console.log(`✓ ${test.op}: ${JSON.stringify(test.input)} → ${JSON.stringify(result)}`);
        passed++;
      } else {
        console.log(`✗ ${test.op}: Expected ${JSON.stringify(test.expected)}, got ${JSON.stringify(result)}`);
        failed++;
      }
    } catch (error) {
      console.log(`✗ ${test.op}: Error - ${error.message}`);
      failed++;
    }
  }
  
  console.log(`\n${passed} passed, ${failed} failed out of ${tests.length} tests`);
  return failed === 0;
}

runTests().then(success => process.exit(success ? 0 : 1));

File: test-transformations.py
#!/usr/bin/env python3
import requests
import json
import time

BASE_URL = "https://textforge.co"

# Test all 28 actual transformations
transformations = [
    # Formatting
    {"action": "slugify", "text": "Hello World", "expected": "hello-world"},
    {"action": "camelcase", "text": "hello world", "expected": "helloWorld"},
    {"action": "snakecase", "text": "Hello World", "expected": "hello_world"},
    {"action": "kebabcase", "text": "Hello World", "expected": "hello-world"},
    {"action": "pascalcase", "text": "hello world", "expected": "HelloWorld"},
    {"action": "constantcase", "text": "hello world", "expected": "HELLO_WORLD"},
    
    # Case
    {"action": "titlecase", "text": "hello world", "expected": "Hello World"},
    {"action": "sentencecase", "text": "hello world. how are you?", "expected": "Hello world. how are you?"},
    
    # Manipulation
    {"action": "reverse", "text": "hello", "expected": "olleh"},
    {"action": "truncate", "text": "hello world this is a long text", "params": {"length": 10}, "expected": "hello worl..."},
    {"action": "removemultiple", "text": "hello    world", "expected": "hello world"},
    {"action": "removespecial", "text": "hello!@#$world", "expected": "helloworld"},
    {"action": "trimtext", "text": "   hello   ", "expected": "hello"},
    
    # Extraction
    {"action": "extractemails", "text": "Contact us at test@example.com or admin@example.org", "expected": ["test@example.com", "admin@example.org"]},
    {"action": "extracturls", "text": "Visit https://example.com and http://test.org", "expected": ["https://example.com", "http://test.org"]},
    {"action": "extractnumbers", "text": "I have 5 apples and 10 oranges", "expected": [5, 10]},
    
    # Counting
    {"action": "countwords", "text": "hello world foo bar", "expected": 4},
    
    # Encoding
    {"action": "base64encode", "text": "hello", "expected": "aGVsbG8="},
    {"action": "base64decode", "text": "aGVsbG8=", "expected": "hello"},
    {"action": "htmlencode", "text": "<div>", "expected": "&lt;div&gt;"},
    {"action": "htmldecode", "text": "&lt;div&gt;", "expected": "<div>"},
    {"action": "markdownplain", "text": "**bold** and *italic*", "expected": "bold and italic"},
    {"action": "unicodenormalize", "text": "café", "expected": "café"},
    
    # Fun
    {"action": "leet", "text": "hello", "expected": "h3ll0"},
    {"action": "morse", "text": "SOS", "expected": "... --- ..."},
    
    # Security/Hashing
    {"action": "hash", "text": "hello", "expected": "2cf24dba5fb0a30e26e83b2ac5b9e29e1b161e5c1fa7425e73043362938b9824"},
    
    # Random (just test it returns a string)
    {"action": "random", "text": "test", "params": {"length": 10}, "expected_type": "string"},
    
    # Palindrome
    {"action": "palindromecheck", "text": "racecar", "expected": {"isPalindrome": True, "normalizedText": "racecar"}},
]

passed = 0
failed = 0

print("Testing all 28 transformations...\n")

for i, test in enumerate(transformations, 1):
    action = test["action"]
    text = test["text"]
    expected = test.get("expected")
    expected_type = test.get("expected_type")
    params = test.get("params", {})
    
    url = f"{BASE_URL}/transform?text={text}&action={action}"
    
    # Add params to URL if present
    if params:
        for key, value in params.items():
            url += f"&{key}={value}"
    
    try:
        response = requests.get(url, timeout=5)
        data = response.json()
        
        if data.get("success"):
            result = data.get("result")
            
            # For random, just check it's a string with right length
            if expected_type:
                if expected_type == "string" and isinstance(result, str):
                    if params.get("length"):
                        if len(result) == params["length"]:
                            print(f"✓ {action}")
                            passed += 1
                        else:
                            print(f"✗ {action} - Wrong length: {len(result)}")
                            failed += 1
                    else:
                        print(f"✓ {action}")
                        passed += 1
                else:
                    print(f"✗ {action} - Wrong type: {type(result)}")
                    failed += 1
            # Compare results
            elif result == expected:
                print(f"✓ {action}")
                passed += 1
            else:
                print(f"✗ {action} - Expected: {expected}, Got: {result}")
                failed += 1
        else:
            print(f"✗ {action} - API Error: {data.get('error')}")
            failed += 1
            
    except Exception as e:
        print(f"✗ {action} - Exception: {str(e)}")
        failed += 1
    
    # Small delay to avoid rate limiting
    if i % 5 == 0:
        time.sleep(0.5)

print(f"\n{'='*50}")
print(f"Results: {passed} passed, {failed} failed out of {len(transformations)} tests")
print(f"{'='*50}")

if failed == 0:
    print("\n🎉 All transformations working correctly!")
    exit(0)
else:
    print(f"\n⚠️  {failed} transformation(s) failed")
    exit(1)
